devito/ir/iet/nodes.py

```diff
@@ -124,20 +124,20 @@ def args_frozen(self):
     def __str__(self):
         return str(self.ccode)
 
-    @abc.abstractproperty
+    @property
     def functions(self):
         """All AbstractFunction objects used by this node."""
-        raise NotImplementedError()
+        return ()
```

Comment:
why the change?

Reply:
lots of subclasses share this method which represents the basic cases. That allowed dropping a large chunk of redundant code across various subclasses